ecommerce/coupons/views.py

```
@@ -108,6 +109,39 @@ def voucher_is_valid(voucher, product, request):
     return True, ''
 
 
+def single_product(product, api, benefit, request):
+    try:
+        course = api.courses(product.course_id).get()
+    except SlumberHttpBaseException as e:
+        logger.exception('Could not get course information. [%s]', e)
+        return {
+            'error': _('Could not get course information. [{error}]'.format(error=e))
```

do not return an error but rather throw an exception.